src/transformers/models/mobilevitv2/modeling_mobilevitv2.py

```diff
@@ -0,0 +1,1200 @@
+        # reduce spatial dimensions if needed
+        if self.downsampling_layer:
+            features = self.downsampling_layer(features)
+
+        # local representation
+        features = self.conv_kxk(features)
+        features = self.conv_1x1(features)
+
+        # convert feature map to patches
+        patches, output_size = self.unfolding_pytorch(
+            features,
+        )
+
+        # learn global representations
+        patches = self.transformer(patches)
+        patches = self.layernorm(patches)
+
+        # convert patches back to feature maps
+        # [B x Patch x Patches x C] --> [B x C x Patches x Patch]
+        features = self.folding_pytorch(patches, output_size)
+
+        features = self.conv_projection(features)
+        return features
+
+
+class MobileViTv2Encoder(nn.Module):
+    def __init__(self, config: MobileViTv2Config) -> None:
+        super().__init__()
+        self.config = config
+
+        self.layer = nn.ModuleList()
+        self.gradient_checkpointing = False
+
+        # segmentation architectures like DeepLab and PSPNet modify the strides
+        # of the classification backbones
+        dilate_layer_4 = dilate_layer_5 = False
+        if config.output_stride == 8:
+            dilate_layer_4 = True
+            dilate_layer_5 = True
+        elif config.output_stride == 16:
+            dilate_layer_5 = True
+
+        dilation = 1
+
+        layer_0_dim = int(
+            make_divisible(
+                bound_fn(min_val=16, max_val=64, value=32 * config.width_multiplier), divisor=8, min_value=16
+            )
+        )
+        layer_1_dim = int(make_divisible(64 * config.width_multiplier, divisor=16))
+        layer_2_dim = int(make_divisible(128 * config.width_multiplier, divisor=8))
+        layer_3_dim = int(make_divisible(256 * config.width_multiplier, divisor=8))
+        layer_4_dim = int(make_divisible(384 * config.width_multiplier, divisor=8))
+        layer_5_dim = int(make_divisible(512 * config.width_multiplier, divisor=8))
+        ffn_multiplier = 2
+
+        # 1: layer_1
+        layer_1 = MobileViTv2MobileNetLayer(
+            config,
+            in_channels=layer_0_dim,
+            out_channels=layer_1_dim,
+            stride=1,
+            num_stages=1,
+        )
+        self.layer.append(layer_1)
+
+        # 2: layer_2
+        layer_2 = MobileViTv2MobileNetLayer(
+            config,
+            in_channels=layer_1_dim,
+            out_channels=layer_2_dim,
+            stride=2,
+            num_stages=2,
+        )
+        self.layer.append(layer_2)
+
+        # 3: layer_3
+        layer_3 = MobileViTv2Layer(
+            config,
+            in_channels=layer_2_dim,
+            out_channels=layer_3_dim,
+            attn_unit_dim=int(make_divisible(128 * config.width_multiplier, divisor=8)),
+            stride=2,
+            ffn_multiplier=ffn_multiplier,
+            n_attn_blocks=2,
+        )
```

Comment:
For all of the `MobileViTv2Layer` layer here and all below: 
* `attn_unit_dim` shouldn't be calculated form hard coded values here e.g. 128 and divisor etc. These values should be part of the configuration e.g. `make_divisible(config.x * config.width_multiplier, divisor=config.y),`
* `stride` argument can be remove - it's 2 for everything
* `n_attn_blocks` should be in the config 
* `patch_size` is taken from the config - arg should be remove and config used instead 
* `ffn_multiplier` can be removed as an arg - its 2 everywhere (and should come from the config)